operations/src/main/java/qonduit/operations/splits/SplitLookupOperation.java

```diff
@@ -0,0 +1,132 @@
+    public static final String SPLIT_FILE_CREATION_ERROR = "Error creating splits file for table: %s";
+    public static final String TABLE_DELETED_ERROR = "Table deleted: %s";
+
+    private static final Logger LOG = LoggerFactory.getLogger(SplitLookupOperation.class);
+
+    private SplitLookupRequest request = null;
+    private ChannelHandlerContext ctx = null;
+    private Server server = null;
+
+    @Override
+    public void run() {
+
+        final SplitLookupResponse response = new SplitLookupResponse();
+        response.setRequestId(this.request.getRequestId());
+        response.setEndOfResults(true);
+
+        if (server.getSplitStore() == null) {
+            response.setErrorMessage(SPLIT_SERVER_DISABLED_ERROR);
+        } else {
+
+            final String tableName = request.getTableName();
+            final String tableId = server.getDataStore().getConnector().tableOperations().tableIdMap().get(tableName);
+
+            if (tableName.startsWith(Namespace.ACCUMULO.name() + ".")) {
+                response.setErrorMessage(String.format(INVALID_TABLE_ERROR, tableName));
+            } else if (tableId == null) {
+                response.setErrorMessage(String.format(TABLE_NOT_FOUND_ERROR, tableName));
+            } else {
+                final String lookup = request.getRow();
+
+                AtomicReference<Path> splitsFile = server.getSplitStore().getSplitsFileForTable(tableName);
+
+                if (splitsFile == null) {
+                    response.setErrorMessage(String.format(NO_SPLITS_FILE_ERROR, tableName));
+                } else if (splitsFile == SplitStore.ERROR_REF) {
+                    response.setErrorMessage(String.format(SPLIT_FILE_CREATION_ERROR, tableName));
+                } else if (splitsFile == SplitStore.TABLE_DELETED_REF) {
+                    response.setErrorMessage(String.format(TABLE_DELETED_ERROR, tableName));
+                } else {
+                    // do lookup
+                    Text lookupRow = MetadataSchema.TabletsSection.encodeRow(TableId.of(tableId), new Text(lookup));
+
+                    Scanner scanner = RFile.newScanner().from(splitsFile.get().toFile().getAbsolutePath())
+                            .withoutSystemIterators().withDataCache(1024).withIndexCache(1024).build();
+
+                    scanner.setRange(new Range(lookupRow, null));
+                    var iter = scanner.iterator();
+                    if (iter.hasNext()) {
+                        Map.Entry<Key, Value> entry = iter.next();
+                        Text endRow = MetadataSchema.TabletsSection.decodeRow(entry.getKey().getRow()).getSecond();
+                        Text prevRow = TabletColumnFamily.decodePrevEndRow(entry.getValue());
+                        response.setBeginRow(prevRow == null ? "null" : prevRow.toString());
+                        response.setEndRow(endRow == null ? "null" : endRow.toString());
+                    }
```

Comment:
Could call `server.getSplitStore().getSplitsFileForTable(tableName)` after this, and if not equal to `splitsFile`, then remove `splitsFile` as the file we just read from is no longer the current file. If old file is not read from for a long time, then it could lead to a bunch of files piling up on disk.